ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/HttpRequestRetry.kt

```diff
@@ -130,37 +182,52 @@ public class HttpRequestRetry internal constructor(configuration: Configuration)
          * Specifies delay logic for retries. The [block] accepts the number of retries
          * and should return the number of milliseconds to wait before retrying.
          */
-        public fun delayMillis(block: (retry: Int) -> Long) {
-            delayMillis = block
+        public fun delayMillis(
+            respectRetryAfterHeader: Boolean = true,
+            block: DelayContext.(retry: Int) -> Long
+        ) {
+            delayMillis = {
+                val retryAfter = response?.headers?.get(HttpHeaders.RetryAfter)?.toLong()
```

Comment:
if `RetryAfter` is the Date format instead of numeric this will cause an exception.  I think it is ok to not handle the Date format of the header (see: https://www.geeksforgeeks.org/http-headers-retry-after/) but the `toLong()` needs protection.